GitBranch in version info is problematic and should be removed

/kind bug

**What steps did you take and what happened:**
1. `git checkout v0.4.5`
1. `make release` which calls `make release-binaries`
...
```
docker run \
                --rm \
                -e CGO_ENABLED=0 \
                -e GOOS=linux \
                -e GOARCH=amd64 \
                -v "$(pwd):/workspace" \
                -w /workspace \
                golang:1.12.10 \
                go build -a -ldflags '-X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitCommit=v0.4.5-dirty' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitTreeState=dirty' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitMajor=0' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitMinor=4' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitVersion=v0.4.5-dirty' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitBranch=(HEAD' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitReleaseTag=v0.4.5' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitReleaseCommit=0feb135482a61c' -extldflags "-static"' \
                -o out/clusterawsadm-linux-amd64 ./cmd/clusterawsadm
bash: -c: line 0: syntax error near unexpected token `('
bash: -c: line 0: `docker run   --rm    -e CGO_ENABLED=0        -e GOOS=linux   -e GOARCH=amd64         -v "$(pwd):/workspace"  -w /workspace   golang:1.12.10  go build -a -ldflags '-X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitCommit=v0.4.5-dirty' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitTreeState=dirty' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitMajor=0' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitMinor=4' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitVersion=v0.4.5-dirty' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitBranch=(HEAD' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitReleaseTag=v0.4.5' -X 'sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitReleaseCommit=0feb135482a61c' -extldflags "-static"'   -o out/clusterawsadm-linux-amd64 ./cmd/clusterawsadm'
make[2]: *** [release-binary] Error 1
```


**What did you expect to happen:**
No error

**Anything else you would like to add:**
The code in `hack/version.sh` runs this to determine the "git branch:

```
git branch | grep \* | cut -d ' ' -f2
```

This is problematic when you have a detached head when you check out a tag such as v0.4.5. The output of `git branch` looks like this:

```
* (HEAD detached at v0.4.5)
  0.4/fix-examples
  1204
  backport_clusterawsadm_version
  ...
```

This results in a bad ldflag: `sigs.k8s.io/cluster-api-provider-aws/cmd/versioninfo.GitBranch=(HEAD`

I suggest we remove GitBranch from version.sh and versioninfo.go. It's not useful and the commit/tag fields we have cover the information we need.

--- a/cmd/versioninfo/versioninfo.go
+++ b/cmd/versioninfo/versioninfo.go
@@ -23,8 +23,6 @@ import (
 )
 
 var (
-	// GitBranch is the branch from which this binary was built
-	GitBranch string
 	// GitReleaseTag is the git tag from which this binary is released
 	GitReleaseTag string
 	// GitReleaseCommit is the commit corresponding to the GitReleaseTag
@@ -60,7 +58,6 @@ func printShortCleanVersionInfo() {
 func printVerboseVersionInfo() {
 	fmt.Println("Version Info:")
 	fmt.Printf("GitReleaseTag: %q, Major: %q, Minor: %q, GitRelaseCommit: %q\n", GitReleaseTag, GitMajor, GitMinor, GitReleaseCommit)
-	fmt.Printf("Git Branch: %q\n", GitBranch)
 	fmt.Printf("Git commit: %q\n", GitCommit)
 	fmt.Printf("Git tree state: %q\n", GitTreeState)
 }

--- a/hack/version.sh
+++ b/hack/version.sh
@@ -67,7 +67,6 @@ version::get_version_vars() {
         fi
     fi
 
-    GIT_BRANCH=$(git branch | grep \* | cut -d ' ' -f2)
     GIT_RELEASE_TAG=$(git describe --abbrev=0 --tags)
     GIT_RELEASE_COMMIT=$(git rev-list -n 1  ${GIT_RELEASE_TAG} | head -c 14)
 }
@@ -91,7 +90,6 @@ version::ldflags() {
     add_ldflag "GitMajor" "${GIT_MAJOR}"
     add_ldflag "GitMinor" "${GIT_MINOR}"
     add_ldflag "GitVersion" "${GIT_VERSION}"
-    add_ldflag "GitBranch" "${GIT_BRANCH}"
     add_ldflag "GitReleaseTag" "${GIT_RELEASE_TAG}"
     add_ldflag "GitReleaseCommit" "${GIT_RELEASE_COMMIT}"
 

